digitalocean/app/resource_app.go

```diff
@@ -314,3 +347,227 @@ func waitForAppDeployment(client *godo.Client, id string, timeout time.Duration)
+		if err != nil {
+			return err
+		}
+		alerts = appAlerts
+		log.Printf("[DEBUG] Got alerts from api %v\n", appAlerts)
+	}
+
+	if hasAppLevelAlerts {
+		err := resourceDigitalOceanAlertDestinationUpdate(appID, alerts, appConfigurationAlert, client, func(a, b *godo.AppAlert) bool {
+			return string(a.Spec.Rule) == string(b.Spec.Rule)
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	if hasComputeComponentAlerts {
+		err := resourceDigitalOceanAlertDestinationUpdate(appID, alerts, computeComponentAlerts, client, func(a, b *godo.AppAlert) bool {
+			return string(a.Spec.Rule) == string(b.Spec.Rule)
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func mapComputeComponentAlert(d *schema.ResourceData) []*godo.AppAlert {
+	spec, ok := d.GetOk("spec")
+	if !ok {
+		return nil
+	}
+
+	specList := spec.([]interface{})
+	if len(specList) == 0 {
+		return nil
+	}
+	specMap, ok := specList[0].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	service, ok := specMap["service"]
+	if !ok {
+		return nil
+	}
+
+	serviceList, ok := service.([]interface{})
+	if !ok {
+		return nil
+	}
+
+	var alertDetails []*godo.AppAlert
+
+	for _, computeServiceMap := range serviceList {
+		// Access the "alert" list within each service
+		computeServiceMapTyped, ok := computeServiceMap.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		alertRaw, ok := computeServiceMapTyped["alert"].([]interface{})
+		if !ok {
+			continue // Skip if no alerts are defined for this service
+		}
+
+		// Iterate over each alert configuration
+		for _, alert := range alertRaw {
+			alertMap, ok := alert.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			notifications, ok := alertMap["notifications"].([]interface{})
+
+			if !ok {
+				continue
+			}
+
+			if len(notifications) == 0 {
+				continue
+			}
+
+			// Create a new godo.AppAlert and populate its fields
+			appAlert := &godo.AppAlert{
+				Spec:          &godo.AppAlertSpec{},
+				Emails:        []string{},
+				SlackWebhooks: []*godo.AppAlertSlackWebhook{},
+			}
+			if rule, ok := alertMap["rule"].(string); ok {
+				appAlert.Spec.Rule = godo.AppAlertSpecRule(rule)
+			}
+			if value, ok := alertMap["value"].(float32); ok {
+				appAlert.Spec.Value = value
+			}
+			if operator, ok := alertMap["operator"].(string); ok {
+				appAlert.Spec.Operator = godo.AppAlertSpecOperator(operator)
+			}
+			if window, ok := alertMap["window"].(string); ok {
+				appAlert.Spec.Window = godo.AppAlertSpecWindow(window)
+			}
+
+			emails, slackWebhooks := extractEmailAndSlackNotifications(notifications)
+			appAlert.Emails = emails
+			appAlert.SlackWebhooks = slackWebhooks
+
+			alertDetails = append(alertDetails, appAlert)
+		}
+	}
+
+	return alertDetails
+}
+
+func extractEmailAndSlackNotifications(notifications []interface{}) ([]string, []*godo.AppAlertSlackWebhook) {
+	notification := notifications[0].(map[string]interface{})
```

Comment:
I see that callers of `extractEmailAndSlackNotifications` ensure the slice length is greater than 0 before calling it, but IMO it's a bit easier to follow the code if the slice check (`if len(notifications) > 0 { ... }`) resides above where the slice is accessed (`notifications[0].(map[string]interface{}`) so it's obvious the zero-length case is handled.

Only mentioning it because the unchecked (`notifications[0]`) stuck out to me.